### ISSUE TITLE
Clickhouse dialect: CREATE USER new_user IDENTIFIED BY 'secret';

### ISSUE THREAD

**Issue by LeviticusNelson** [sqlfluff/sqlfluff#7157]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The `IDENTIFY BY` keywords are not able to be parsed in the clickhouse dialect.

### Expected Behaviour

These rules should be valid:
https://clickhouse.com/docs/sql-reference/statements/create/user#identification

### Observed Behaviour

`Line 2, Position 25: Found unparsable section: "IDENTIFIED BY 'secret';\n\nGRANT INS..."`

### How to reproduce

in .sql:
```sql
-- sqlfluff:dialect:clickhouse
CREATE USER new_user IDENTIFIED BY 'secret';
```

### Dialect

clickhouse

### Version

sqlfluff - 3.4.2
python - 3.13.7

### Configuration

N/A

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by LeviticusNelson** [comment]
#7158
